tests/lib.rs

```diff
@@ -0,0 +1,700 @@
+use async_memcached::{Client, Error, Status};
+use rand::seq::IteratorRandom;
+
+// Note: Each test should run with keys unique to that test to avoid async conflicts.  Because these tests run concurrently,
+// it's possible to delete/overwrite keys created by another test before they're read.
```

Comment:
If tests are run in serial then it shouldn't matter, but when running things in parallel I'm not sure if there's much of a way around it (without mocking) since we're interacting with a single server.  We want to start from a known state for each test, so the setup ensures that keys are configured on the server as expected.  One of the consequences of running things in parallel is that you can potentially have cross-effects if the same keys are used in many tests.